pkg/client/unversioned/helper.go

```diff
@@ -258,16 +258,35 @@ func SetKubernetesDefaults(config *restclient.Config) error {
 	if config.APIPath == "" {
 		config.APIPath = legacyAPIPath
 	}
-	g, err := registered.Group(api.GroupName)
-	if err != nil {
-		return err
+	if config.GroupVersion == nil || !registered.IsEnabledVersion(unversioned.GroupVersion{Group: api.GroupName, Version: config.GroupVersion.Version}) {
+		g, err := registered.Group(api.GroupName)
+		if err != nil {
+			return err
+		}
+		copyGroupVersion := g.GroupVersion
+		config.GroupVersion = &copyGroupVersion
 	}
-	// TODO: Unconditionally set the config.Version, until we fix the config.
-	copyGroupVersion := g.GroupVersion
-	config.GroupVersion = &copyGroupVersion
 	if config.NegotiatedSerializer == nil {
 		config.NegotiatedSerializer = api.Codecs
 	}
-
 	return restclient.SetKubernetesDefaults(config)
 }
+
+func setGroupDefaults(groupName string, config *restclient.Config) error {
+	config.APIPath = defaultAPIPath
+	if config.UserAgent == "" {
+		config.UserAgent = restclient.DefaultKubernetesUserAgent()
+	}
+	if config.GroupVersion == nil || !registered.IsEnabledVersion(unversioned.GroupVersion{Group: groupName, Version: config.GroupVersion.Version}) {
```

Comment:
With only nil check I'm getting following warnings during describe:
```
$ kubectl describe scheduledjob/pi
W0725 20:24:15.733731   19649 request.go:341] Field selector: batch/v2alpha1 - events - involvedObject.kind - ScheduledJob: need to check if this is versioned correctly.
W0725 20:24:15.733778   19649 request.go:341] Field selector: batch/v2alpha1 - events - involvedObject.name - pi: need to check if this is versioned correctly.
W0725 20:24:15.733783   19649 request.go:341] Field selector: batch/v2alpha1 - events - involvedObject.namespace - default: need to check if this is versioned correctly.
W0725 20:24:15.733790   19649 request.go:341] Field selector: batch/v2alpha1 - events - involvedObject.uid - aacdaafd-52a5-11e6-be6a-525400f25e34: need to check if this is versioned correctly.
Name:				pi
Namespace:			default
Schedule:			* * * * * ?
Concurrency Policy:		Allow
Suspend:			False
Starting Deadline Seconds:	<unset>
Image(s):			perl
Selector:			<unset>
Parallelism:			<unset>
Completions:			<unset>
No volumes.
Labels:			<none>
Last Schedule Time:	<unset>
Active Jobs:		<none>
No events.
```
These are caused by following requests:
```
GET http://localhost:8080/api/batch/v2alpha1/namespaces/default/events?fieldSelector=involvedObject.kind%3DScheduledJob%2CinvolvedObject.name%3Dpi%2CinvolvedObject.namespace%3Ddefault%2CinvolvedObject.uid%3Daacdaafd-52a5-11e6-be6a-525400f25e34
```

The best option would be to verify if `GroupVersion` is nil or is targeting different group (remember there's only one GV in that config), in which case we'll set defaults appropriately. @smarterclayton wdyt?